### ISSUE TITLE
Climate Normal Years

### ISSUE THREAD

**Issue by aarony2j** [JGCRI/gcamdata#587]
In the old energy code for LA144.building_det_en.R, there are these few lines that refer to `X_climate_normal_years`, where `climate_normal_years` is `1981:2000`.

Any idea why this is the case? It'd be nice to include some description. (And @bpbond asked about it here: #573). Thanks!

```
#Subset the thermal services
L144.in_EJ_ctry_bld_thrm_F <- subset( L144.in_EJ_ctry_bld_serv_F, service %in% thermal_services )
L144.in_EJ_ctry_bld_thrm_F$variable <- hddcdd_mapping$variable[ match( L144.in_EJ_ctry_bld_thrm_F$service, hddcdd_mapping$service ) ]

#Then, calculate the "normals" from the HDD and CDD data, both at the country level and the GCAM 3.0 region level.
L143.HDDCDD_scen_ctry_Y$normal <- apply( L143.HDDCDD_scen_ctry_Y[ X_climate_normal_years ], 1, mean )

##NOTE: wherever the climate normal is less than 1, this will round to 0 when hddcdd are read in to the model. Go ahead and put these at 0,
# in order to remove any instances where HDDCDD are treated as being greater than 0 when the model will see a 0
L143.HDDCDD_scen_ctry_Y$normal[ L143.HDDCDD_scen_ctry_Y$normal < 1 ] <- 0

#Calculate the normals for each country over the historical time period
L143.HDDCDD_scen_RG3_Y$normal <- apply( L143.HDDCDD_scen_RG3_Y[ X_climate_normal_years ], 1, mean )
L144.in_EJ_ctry_bld_thrm_F$normal <- L143.HDDCDD_scen_ctry_Y$normal[
      match( vecpaste( L144.in_EJ_ctry_bld_thrm_F[ c( "iso", "variable" ) ] ),
             vecpaste( L143.HDDCDD_scen_ctry_Y[ c( "iso", "variable" ) ] ) ) ]
```

**Comment by pkyle** [comment]
The reason for the years chosen is that for the SRES scenarios, 2000 was the base year, and the future was anything from 2001 onward. Of course in the GCMs there's really no difference between historical and future years, but I figured that the models must have done some sort of calibration to replicate observational data, at least on some scale, for the historical years. Maybe that's not right? I don't know. Anyway, what would be ideal for calibration is re-analysis data that incorporates observations, but when we tried to use that, we got HUGE jumps in HDD/CDD in going from present to future. Using the averaged (i.e., smoothed) historical year data from a "future" GCM run does take care of that.
The basic problem with the method of course is that this is used for our historical estimation of energy consumption by country and service, and it's the sort of thing that we wouldn't want to be dependent on the choice of a GCM. What we should probably do is use re-analysis data for the historical years, calculate climate normals from that, and then in some way index the future output to the historical data in the base years (e.g., 2005-2010).
Note that the building service functions are driven by multiplicative changes. A region going from 3000 HDD in 2010 to 3500 HDD in 2015 isn't going to rock the boat a whole lot, but a region going from 5 HDD to 500 HDD will have a 100-times increase in heating service demands.
